### DISCUSSION TITLE
use Golang to create a new input plugin

### DISCUSSION THREAD

**Discussion by 666MrFang** [fluent/fluent-bit#6427] [Q&A]
Ask a question, how to use Golang to create a new input plugin, is there any example? I currently only see examples of using Golang to create output plugins in fluent/fluent-bit-go

**Comment by patrick-stephens** [comment]
I mentioned this at the community meeting today: https://github.com/calyptia/lts-advanced-plugin-dummy

**Reply by 666MrFang** [reply]
Sir，is calyptia/plugin a guide document for creating input plugins in Golang officially adopted by fluent? Is there any official documentation for fluent to specify the creation of input plugins through Golang?

**Reply by patrick-stephens** [reply]
Calyptia are the maintainers of Fluent Bit, this is an example of creating an input plugin in case you wanted to use it: it's a simple skeleton you can reuse if needs be - we use it internally for other plugins for our Enterprise offerings for example.

As you say though, we probably need some docs around this in as well so I'll see if I can flag that up to get something put together.
